File: Go/jobspls/algorithms/frog_jump.go
package main

import (
	"fmt"
	"math"
)

// How many jumps does it take to reach dest by dist?
func FrogJump(start, dest, dist uint) (uint, error) {
	if start > dest {
		return 0, fmt.Errorf("start > dest")
	}

	diff := dest - start
	jumps := uint(diff/dist) + uint(math.Min(1.0, float64(diff%dist)))

	return jumps, nil
}

File: Java/jobspls/algorithms/FrogJump.java
package jobspls.algorithms;

import java.lang.Math;
import java.util.OptionalInt;

public class FrogJump {
    public static OptionalInt frog_jump(int start, int dest, int dist) {
        if (
            start > dest ||
            start < 0 ||
            dest < 0 ||
            dist <= 0
        ) {
            return OptionalInt.empty();
        }

        var diff = dest - start;
        var jumps = (int)(diff/dist) + (int)(Math.min(1.0, diff%dist));

        return OptionalInt.of(jumps);
    }
}
